slycot/tests/test_tb05ad.py

```diff
@@ -187,6 +187,55 @@ def test_tb05ad_resonance(self):
             transform.tb05ad(2, 1, 1, jomega, A, B, C, job='NH')
         assert cm.exception.info == 2
 
+    def test_tb05ad_balance(self):
+        """Test balancing in tb05ad.
+
+        Tests for the cause of the problem reported in issue #11
+        balancing permutations were not correctly applied to the
+        C and D matrix.
+        """
+
+        # find a good test case. Some sparsity,
+        # some zero eigenvalues, some non-zero eigenvalues,
+        # and proof that the 1st step, with dgebal, does some
+        # permutation
+        crit = False
+        while not crit:
+            A = np.random.randn(8, 8)
+            A[np.random.uniform(size=(8, 8)) > 0.35] = 0.0
+
+            Aeig = eig(A)[0]
+            neig0 = np.sum(np.abs(Aeig) == 0)
+            As, T = matrix_balance(A)
+            nperm = np.sum(np.diag(T == 0))
+
+            crit = nperm < 8 and nperm >= 4 and \
+                neig0 > 1 and neig0 <= 3
+
+        # print("number of permutations", nperm, "eigenvalues=0", neig0)
+        B = np.random.randn(8, 4)
+        C = np.random.randn(3, 8)
+
+        # do a run
+        jomega = 1.0
+        At, Bt, Ct, rcond, g_jw, ev, hinvb, info = transform.tb05ad(
+            8, 4, 3, jomega, A, B, C, job='AG')
+
+        # remove information on Q, in lower sub-triangle part of A
+        At = np.triu(At, k=-1)
+
+        # now after the balancing in DGEBAL, and conversion to
+        # upper Hessenberg form:
+        # At = Q^T * (P^-1 * A * P ) * Q
+        # with Q orthogonal
+        # Ct = C * P * Q
+        # Bt = Q^T * P^-1 * B
+        # so test with Ct * At * Bt  ==  C * A * B
+        # and verify that eigenvalues of both A matrices are close
+        assert_almost_equal(np.dot(np.dot(Ct, At), Bt),
+                            np.dot(np.dot(C, A), B))
+        assert_almost_equal(eig(At)[0], eig(A)[0])
```

Comment:
Is the order and scaling of eigenvalues guaranteed? Could run into assertion failures on some build systems. Or do `scipy.linalg.eig()` and MB05AD call the same LAPACK code and thus always yield the same results?

Reply:
I'll put in some sorting.

Reply:
What about scaling, including different signs? I had to deal with that in `test_mb05md()` (#79)

Reply:
Now you're confusing me. Do you mean scaling of the eigenvalues? But then the frequency should also be scaled to calculate the response, and I don't see anything on that in the TB05AD.f code. Eigenvectors may be differently scaled, but I did not do a test on that. 

Or do you mean the scaling in the matrix_balance / dgebal step. I now added some code for that, got to the same testing case as before, on the 10th matrix tried.

Reply:
Okay strike that. Brainfart by me. Mixed up eigenvalue properties with eigenvectors.